### PR TITLE
typescript!!!!

### PR DESCRIPTION
**Pull request by cerqiest**
Convert everything over to TS.

### CODE REVIEW

**Comment by cerqiest** [comment]
I'm confused about what my `module` and `target` settings should be here and in tsi. @blockarchitech please review when you get a chance

File: out/app.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const config = {
+    port: process.env.PORT || 3000,
+    cookieMaxAge: 1000 * 60 * 60 * 24 * 7,
+    routes: {
+        "/": indexRouter,
+    },
+};
+// Scaffold app
+const app = express();
+app.disable("x-powered-by");
```
Comment (blockarchitech) [inline out/app.js:22]: Please replace this with a valid header. Some browsers will block CORS traffic.

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "outDir": "./out",
+    "moduleResolution": "NodeNext"
+  },
+
+  "include": ["src/**/*.*"],
+  "exclude": ["node_modules", "<node_internals>/**"],
+
+  "$schema": "https://json.schemastore.org/tsconfig",
```
Comment (blockarchitech) [inline tsconfig.json:16]: 16 and 17 are unnecessary

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "compilerOptions": {
+    "target": "ES2022",
```
Comment (blockarchitech) [inline tsconfig.json:3]: ES2020 is a good target

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "module": "ES2022",
```
Comment (blockarchitech) [inline tsconfig.json:4]: 99% of the time I'd use CJS, but ES2020 would work.

**Review by blockarchitech — CHANGES_REQUESTED** [review]
Overall fine. Please see comments.

File: out/app.js (hunk 1 above)
Comment (cerqiest) [reply out/app.js:22]: What does this have to do with CORS? I'm removing the `X-Powered-By: Express` header.

Comment (blockarchitech) [reply out/app.js:22]: CORS can sometimes use the webserver

Comment (cerqiest) [reply out/app.js:22]: What does that have to do with the `X-Powered-By` header?

Comment (blockarchitech) [reply out/app.js:22]: This and `Server` is how CORS detects that. It is also bad practice to disable headers for almost zero apparent reason.

I also am not sure why simple things like this need to be fought over.

Comment (cerqiest) [reply out/app.js:22]: Huh? Could you explain exactly how the header has anything to do with CORS?

Comment (blockarchitech) [reply out/app.js:22]: I have just told you.

I have a pushed a commit to fix this.

**Review by blockarchitech — APPROVED** [review]

**Pull request merged**